test/integration/discovery.go

```diff
@@ -157,6 +288,129 @@ var _ = Describe("Discover Cluster", func() {
 	})
 })
 
+func findEntities(vals []*proto.EntityDTO, condition func(v *proto.EntityDTO) bool, howMany int) []*proto.EntityDTO {
+	found := []*proto.EntityDTO{}
+	for _, v := range vals {
+		if howMany == 0 {
+			break
+		}
+
+		if condition(v) {
+			found = append(found, v)
+			howMany--
+		}
+	}
+	return found
+}
+
+func findGroups(vals []*proto.GroupDTO, condition func(v *proto.GroupDTO) bool, howMany int) []*proto.GroupDTO {
+	found := []*proto.GroupDTO{}
+	for _, v := range vals {
+		if howMany == 0 {
+			break
+		}
+
+		if condition(v) {
+			found = append(found, v)
+			howMany--
+		}
+	}
+	return found
+}
+
+func findCommoditiesBought(vals []*proto.EntityDTO_CommodityBought, condition func(v *proto.EntityDTO_CommodityBought) bool, howMany int) []*proto.EntityDTO_CommodityBought {
+	found := []*proto.EntityDTO_CommodityBought{}
+	for _, v := range vals {
+		if howMany == 0 {
+			break
+		}
+
+		if condition(v) {
+			found = append(found, v)
+			howMany--
+		}
+	}
+	return found
+}
+
+func findNodeConditions(vals []corev1.NodeCondition, condition func(v corev1.NodeCondition) bool, howMany int) []corev1.NodeCondition {
```

Comment:
we already have a function `NodeIsReady` in the file `node_util.go` which can help check if a node is in the ready status or not, maybe you can just use that one